src/models/ComponentStyle.js

```diff
@@ -28,17 +60,37 @@ export default (nameGenerator: NameGenerator, flatten: Flattener, stringifyRules
      * Returns the hash to be injected on render()
      * */
     generateAndInjectStyles(executionContext: Object, styleSheet: StyleSheet) {
+      const { isStaticRules, isInjected, staticGeneratedClassName } = this
+
+      if (isStaticRules && staticGeneratedClassName && isInjected) {
```

Comment:
`isInjected` is generally unnecessary, since it can be inferred. If `staticGeneratedClassName` is set, we can assume that it was injected.

`staticGeneratedClassName` is a bit of an odd name as well. Call it `lastClassName` and just set it every time. Setting it won't be more expensive than a branch i.e some if statements

Instead of evaluating the truthiness of `staticGeneratedClassName`, we can check whether it's not `undefined` (assuming you remove the initialisation to `null` above of course). That won't be as expensive as a cast to a boolean value.